test/sequential/test-deprecation-flags.js

```diff
@@ -32,7 +47,64 @@ execFile(node, traceDep, function(er, stdout, stderr) {
   assert.equal(stdout, '');
   var stack = stderr.trim().split('\n');
   // just check the top and bottom.
-  assert.equal(stack[0], 'Trace: util.p: Use console.error() instead');
+  assert.equal(stack[0], traceMessage);
   assert.equal(stack.pop(), '\'This is deprecated\'');
   console.log('trace ok');
 });
+
+execFile(node, traceDepMultiple, function(er, stdout, stderr) {
+  console.error('--trace-deprecation: deprecation message printed 3 times');
+  assert.equal(er, null);
+  assert.equal(stdout, '');
+
+  var messagesCount = 0;
+  var fileNameCount = 0;
+  var fileNameWithLine = `deprecated-multiple-times.js:${fileNameCount + 1}`;
+
+  stderr.trim().split('\n').forEach(function(line, idx) {
+    if (idx % 12 === 0 && line === traceMessage) {
+      messagesCount += 1;
+    } else if (line.indexOf(fileNameWithLine) > -1) {
+      fileNameCount += 1;
+      fileNameWithLine = `deprecated-multiple-times.js:${fileNameCount + 1}`;
+    }
+  });
+
+  // because we invoked the same function three times
+  assert.equal(messagesCount, 3);
+  assert.equal(messagesCount, fileNameCount);
+  assert.equal(fileNameWithLine, 'deprecated-multiple-times.js:4');
+
+  console.log('trace ok');
+});
+
+execFile(node, traceDepMultipleSameLine, function(er, stdout, stderr) {
+  console.error('--trace-deprecation: deprecation message printed only once');
+  assert.equal(er, null);
+  assert.equal(stdout, '');
+
+  var count = stderr.trim().split('\n').filter(function(currentLine) {
+    return currentLine === traceMessage;
+  }).length;
+
+  // because we invoked the same function two times, but in the same line
+  assert.equal(count, 1);
+
+  console.log('trace ok');
+});
+
+execFile(node, traceDepParen, function(er, stdout, stderr) {
+  console.error('--trace-deprecation: deprecation with paren in the file name');
+
+  // The purpose of this test is to make sure that require('internal/util')'s
+  // `deprecate` function doesn't fail with the files with parenthesis in name.
+  assert.equal(er, null);
+  assert.equal(stdout, '');
+
+  var stack = stderr.trim().split('\n');
+  // just check the top and bottom.
+  assert.equal(stack[0], traceMessage);
+  assert.equal(stack.pop(), '\'This is deprecated\'');
+  console.log('trace ok');
+});
+
```

Comment:
I'm talking about the final line of this file. git apply also complains: `new blank line at EOF`